contracts/src/LightClient.sol

```diff
@@ -105,65 +125,74 @@ contract LightClient {
         BN254.validateScalarField(newState.stakeTableSchnorrKeyComm);
         BN254.validateScalarField(newState.stakeTableAmountComm);
 
-        // check plonk proof
-        // TODO: (alex) replace the vk with the correct one
-        IPlonkVerifier.VerifyingKey memory vk = VkTest.getVk();
-        uint256[] memory publicInput = preparePublicInput(newState, isNewEpoch);
-        PlonkVerifier.verify(vk, publicInput, proof, bytes(""));
-
-        // upon successful verification, update state.
-        // If the newState is in a new epoch, only then should we increment the `currentEpoch`, and
-        // update the stake table. The `finalizedState` (before update) should have the
-        // `epochEndingBlockHeight`
+        // If the newState is in a new epoch, increment the `currentEpoch`, update the stake table.
```

Comment:
Do we assume that this function is called at least once per epoch?

Reply:
yes, so we enforce that the last block of each epoch is must submitted before going to the next epoch:

```solidity
```

Reply:
Makes sense.